### ISSUE TITLE
tracked changes are not accepted in other output formats than docx

### ISSUE THREAD

**Issue by tolot27** [jgm/pandoc#4301]
The Pandoc Docx reader and writer supports track changes of MS Word (command line parameter `--track-changes=accept|reject|all`).

According to the [manual](https://pandoc.org/MANUAL.html) the option `accept` is the default
and means: "inserts all insertions and ignores all deletions." Unfortunately, this is not true for other writers than docx because it is simply not applied. Instead, the spans introduced by the docx reader (if track changing was enabled), clutters the output.

IMHO, this is a documentation bug which could be fixed easily. But it would be better, if track changes will be accepted per default by any writer or at a higher intermediate stage.

Indeed, this could be implemented as a lua filter as already [mentioned](https://groups.google.com/d/msg/pandoc-discuss/fAxWiJyCSoA/x5jBFZ5RAAAJ) and as I did it now with [track-changes-final.lua](https://github.com/tolot27/lua-filters/blob/track-changes/track-changes). But IMHO, it is more user-friendly, if it were included in pandoc natively.

A `sample.md` and `expected_final.markdown` is provided in the [track-changes](https://github.com/tolot27/lua-filters/tree/track-changes/track-changes) repository.

**Comment by jkr** [comment]
I'm not sure I understand. Do you mean "this is not true for other *readers* than docx"? The documentation states that this command only affects the docx reader.

**Comment by jkr** [comment]
If not, do you mean that the markdown writer inserts spans when converting `docx -> markdown` while `--track-changes=accept`? That would be a bug, but I'm pretty certain that's not the case.

**Comment by tolot27** [comment]
#1562 adds track-changes support to the docx writer as well, even if it is not documented yet.
Nevertheless, markdown documents formerly produced by the docx reader (and maybe from someone else) with track-changes enabled, contains the track changes spans. These should be removed by acceptance in case of conversion to other output formats.

**Comment by jkr** [comment]
The spans are not produced by `--track-changes=accept`. They are produced by `--track-changes=all`, which was intended to be messy, and to give information to filters. `--track-changes=accept|reject` produce no spans in the output. This is explained in the documentation:

    Specifies what to do with insertions, deletions, and comments
    produced by the MS Word "Track Changes" feature.  `accept` (the
    default), inserts all insertions, and ignores all
    deletions. `reject` inserts all deletions and ignores
    insertions. Both `accept` and `reject` ignore comments. `all` puts
    in insertions, deletions, and comments, wrapped in spans with
    `insertion`, `deletion`, `comment-start`, and `comment-end`
    classes, respectively.

You're right that the docx writer function should be documented, and it might be nice if other writers could make use of it as well (though this is probably a better job for filters, since it's not native to those formats). But the documentation of the feature is accurate.

**Comment by tolot27** [comment]
The [online manual](https://pandoc.org/MANUAL.html) and the man pages are different to your pasted text. Why?
There is also written: `Both accept and reject ignore comments.` Hence, at least comments remain if docx -> markdown conversion was made.

Nevertheless, it is useful (at least for me), to convert the tracked changes to other output formats (i. e. LaTeX, HTML), too. That makes a review process easier and is the reason why I wrote the track-changes lua filter.

**Comment by jkr** [comment]
Mathias Walter <notifications@github.com> writes:

> The [online manual](https://pandoc.org/MANUAL.html) and the man pages
> are different to your pasted text. Why?

My pasted text is directly from the online manual. What differences do
you see?

> There is also written: `Both accept and reject ignore comments.`
> Hence, at least comments remain if docx -> markdown conversion was
> made.

No -- that means that the reader *ignores* comments. It doesn't read
them.

Have you tried these things out with a conversion from docx?

> Nevertheless, it is useful (at least for me), to convert the tracked
> changes to other output formats (i. e. LaTeX, HTML), too. That makes a
> review process easier and is the reason why I wrote the track-changes
> lua filter.

Great -- as the manual says, writing filters/scripts is why
`--track-changes=all` is in there. But I don't think it makes sense to
have an official conversion to other formats (except may ODT). The
insertion/deletion spans are a hack meant to match docx, to give script
writers as much information as possible. I don't think it should be
shoehorned in to pandoc markdown, as the official way to write edits in
markdown.

**Comment by tolot27** [comment]
>> The [online manual](https://pandoc.org/MANUAL.html) and the man pages
>> are different to your pasted text. Why?
>
>My pasted text is directly from the online manual. What differences do
>you see?

Sorry, my fault. It was at midnight.

> Have you tried these things out with a conversion from docx?

Yes, I tried to convert some docx, but always with `--track-changes=all`.

I got a markdown document (mail, commit) with track-changes included and want to convert it to PDF. But you are right, I can always convert it back to docx and again to markdown or any other format with `pandoc -s -t docx document.md -o - | pandoc -s -t markdown -f docx`. IMHO: that should be documented somewere.

My original misunderstanding was the assumption that `accept (the default)` is the default for all writers which would still makes sense, IMHO. That was the reason to create this issue.

> But I don't think it makes sense to have an official conversion to other formats (except may ODT).

I fully agree, but "*except* may ODT" ... and may HTML (#1560). ;-) As long as there is no "official conversion", I'll implement it in my filter.
